internal/metrics/metrics.go

```diff
@@ -0,0 +1,32 @@
+package metrics
+
+import (
+	"github.com/RHEnVision/provisioning-backend/internal/kafka"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	metricNameSourceAvailabilityCheckRequestTotal = "source_availability_check_request_total"
+)
+
+var TotalAvailabilityCheckReqs = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name:        metricNameSourceAvailabilityCheckRequestTotal,
+		Help:        "availability Check requests count partitioned by type (aws/gcp/azure), source status, component and error",
+		ConstLabels: prometheus.Labels{"service": "provisioning"},
+	},
+	[]string{"type", "status", "component", "error"},
+)
+
+func IncTotalAvailablilityCheckReqs(provider models.ProviderType, component string, StatusType kafka.StatusType, err error) {
+	errString := ""
+	if err != nil {
+		errString = err.Error()
```

Comment:
I'd be fine if this is just `"error"` for now, so we can identify errored requests, we can improve to best practice labeling in follow ups :)

Reply:
So for first version is it ok to keep false/true as I mentioned on slack?

Reply:
yup :)